pkg/pillar/cmd/zedagent/attesttask.go

```diff
@@ -447,13 +460,36 @@ func encodeEventLog(attestCtx *attestContext, quoteMsg *attest.ZAttestQuote) err
 		if event.Index == 8 || event.Index == 9 {
 			tpmEventLog.EventDataString = string(event.Data)
 		} else {
-			tpmEventLog.EventDataString = "Not Applicable"
+			tpmEventLog.EventDataString = notApplicableString
 		}
 		quoteMsg.EventLog = append(quoteMsg.EventLog, tpmEventLog)
 	}
 	return nil
 }
 
+//deduplicateEventLog removes duplicates String/Binary
+func deduplicateEventLog(quoteMsg *attest.ZAttestQuote) {
+	for _, el := range quoteMsg.EventLog {
+		if el.EventDataString != notApplicableString {
+			//we have string representation
+			el.EventDataBinary = nil
+		} else {
+			//remove unneeded not applicable string
+			el.EventDataString = ""
+		}
+	}
+}
+
+//cleanupEventLog removes items from EventLog which exceed size limit
+func cleanupEventLog(quoteMsg *attest.ZAttestQuote) {
+	for i := 0; i < len(quoteMsg.EventLog); i++ {
+		if len(quoteMsg.EventLog[i].EventDataBinary) > eventLogBinarySizeLimit {
+			quoteMsg.EventLog = append(quoteMsg.EventLog[:i], quoteMsg.EventLog[i+1:]...)
```

Comment:
Don't remove the entry; we will need it with the digest.
But you can clear the EventDataBinary field when it exceeds the size.
FWIW I haven't yet found (in https://trustedcomputinggroup.org/resource/pc-client-specific-platform-firmware-profile-specification/) where the eventType goes in what we retrieve from Linux. But clearing huge EventDataBinary is a reasonable step. 

Maybe we should add a EventDataLen field which we can set to len() so that one can see in the controller that a huge thing was dropped.

Reply:
Done. Also I add EventType filling to have more info